### ISSUE TITLE
Selecting spots in image seems to not work if they are transformed...

### ISSUE THREAD

**Issue by constantinpape** [mobie/spatial-transcriptomics-example-project#22]
... with an affine transformation.

Example project: `/g/kreshuk/data/marioni/shila/mouse-atlas-2020/ngff`, view: `stitched-full`.

@tischi: this is a massive view (40 big spot tables, takes 7 minutes to load), so not ideal for debugging....
I will see that I create something smaller for you for debugging soon.

**Comment by tischi** [comment]
You mean it moves to a wrong position?

Is that a merged grid or a transformed grid?

And are you selecting spots in the image or in the table?

And, yes, small example always is good :)

**Comment by constantinpape** [comment]
> You mean it moves to a wrong position?

No, selection of the spot in the image doesn't work at all; the spot is not selected. The other way round (selection in the table) works.

> Is that a merged grid or a transformed grid?

None of the above. It's affine transform to place irregular tiles. (See the video I shared with you in mattermost)

> And are you selecting spots in the image or in the table?

Selection in table works, image doesn't work.

> And, yes, small example always is good :)

Sure, will do.

**Comment by tischi** [comment]
Probably I don't need an example then. Let me have a look first.

**Comment by tischi** [comment]
I think spot selection does not work at all (also if not transformed).

**Comment by constantinpape** [comment]
It works for me in the spatial transcriptomics example project:

https://user-images.githubusercontent.com/4263537/196381454-f9bcdab8-8d94-434b-9fa7-ba0d483e151d.mp4

**Comment by tischi** [comment]
OK thanks! Then I need to have a closer look.

Note to self: I found out that the voxel coordinates within the image upon clicking are returned rounded to the next integer. For a spot image this is not precise enough (normally this is fine because images live on a voxel grid, this is not the case for the spot image). Weird that this should only happen for transformed images.

**Comment by tischi** [comment]
Fixed now by the discussed changes.
